### PR TITLE
Feature lp 22 filter by tag

### PR DESCRIPTION
**Pull request by Charvit123**
## What

Please include a summary of the changes.

- [x] filter course api

# Checklist

- [ ] My code follows the style guidelines of this project and is properly formatted
- [ ] I have performed a self-review of my own changes
- [ ] All existing tests pass, and I have added new ones if required

### CODE REVIEW

File: backend/src/course/course.controller.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { Course } from '@prisma/client';
+import { CourseService } from './course.service';
+
+@UseGuards(JwtAuthGuard)
+@Controller('course')
+export class CourseController {
+  constructor(private readonly courseService: CourseService) {}
+
+  @Get('/')
+  async getAll(): Promise<Course[]> {
```
Comment (AshPrajapati) [inline backend/src/course/course.controller.ts:12]: await is required

File: backend/src/tag/tag.controller.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { Tag } from '@prisma/client';
+import { TagService } from './tag.service';
+
+@UseGuards(JwtAuthGuard)
+@Controller('tag')
+export class TagController {
+  constructor(private readonly tagService: TagService) {}
+
+  @Get('/')
+  async getAll(): Promise<Tag[]> {
```
Comment (AshPrajapati) [inline backend/src/tag/tag.controller.ts:12]: await is required

**Review by AshPrajapati — APPROVED** [review]
Looks Good

File: backend/src/course/course.controller.ts (hunk 1 above)
Comment (Charvit123) [reply backend/src/course/course.controller.ts:12]: done

File: backend/src/tag/tag.controller.ts (hunk 2 above)
Comment (Charvit123) [reply backend/src/tag/tag.controller.ts:12]: done

**Pull request merged**